lib/mozart_fetcher/http_client.ex

```diff
@@ -13,26 +13,28 @@ defmodule HTTPClient do
         hackney: [pool: :origin_pool]
       ]
 
-      case make_request(sanitise(endpoint), headers, options) do
-        {:error, %HTTPoison.Error{reason: "closed"}} ->
-          handle_response(make_request(sanitise(endpoint), headers, options))
-        {k, response} -> handle_response({k, response})
+      case make_request(sanitise(endpoint), headers, options, client) do
```

Comment:
I tried to turn this into this, but the scope of the resp inside the if statement was not the same:
```ex
  if retry_if_unsuccessful?({k, resp}) do
    {k, resp} = make_request(sanitise(endpoint), headers, options, client)
  end

  {k, resp}
end

...

defp retry_if_unsuccessful?({:error, %HTTPoison.Error{reason: :closed}}), do: true

defp retry_if_unsuccessful?({:error, %HTTPoison.Error{reason: :timeout}}), do: true

defp retry_if_unsuccessful?(_), do: false
```